File: spring-web-java/src/main/java/spring/web/java/domain/Token.java
package spring.web.java.domain;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

@Getter
@RedisHash(value = "token", timeToLive = 604800)
public class Token {

    @Id
    private String account;
    private String refreshToken;

    public static Token create(String account, String refreshToken) {
        Token token = new Token();

        token.account = account;
        token.refreshToken = refreshToken;

        return token;
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/Token.kt
package spring.web.kotlin.domain

import org.springframework.data.redis.core.RedisHash

@RedisHash(value = "token", timeToLive = 604800)
class Token private constructor(
    val account: String,
    val refreshToken: String
){
    companion object {
        fun create(account: String, refreshToken: String) {
            Token(
                account = account,
                refreshToken = refreshToken
            )
        }
    }
}